scripts/ts-compiler.js

```diff
@@ -0,0 +1,10 @@
+const ts = require('typescript');
+const tsNodeSwcCompiler = require('ts-node/dist/compilers/swc');
+const typescriptCachedTranspile = require('typescript-cached-transpile/create');
+const compiler = typescriptCachedTranspile.create({
+  compiler: {
+    ...ts,
+    ...tsNodeSwcCompiler
```

Comment:
Just to be sure, if `swc` isn't ready (because the artifacts haven't been installed), it'll fallback to the regular TS compiler?

Reply:
I had not designed it to do that, but I could add that.  Also, I am not
sure how it internally chooses a version to use.  It may fallback to its
WebAssembly build automatically.

Reply:
If there's a wasm build it's fine, I just want to make sure that people who close the repo can still run `yarn install` (otherwise they wouldn't even be able to build the artifacts 😄)

Reply:
Good point, I hadn't considered that.  I will test it and reply once I'm certain it will work.

Reply:
After working on https://github.com/yarnpkg/berry/pull/1939 I don't think native binaries is an option at all and that we should only use the WASM version, even if it's a bit slower

Reply:
swc's native component runs within the node process.  Does that fall in the same bucket as native binaries, as far as #1939 is concerned?  Reading the linked issue, it talks about external executables being blocked, but it's unclear if dlls are blocked.  Is your assessment that we should avoid native dlls entirely, because we can't be sure that corporate security software will allow it?

I filed an issue with swc about the best way to use their wasm build via @swc/core.  We could use @swc/wasm directly, but it seems more likely that swc wants to expose a proper way to do this via @swc/core.

Reply:
Sorry for the really late response, I don't really want anything native at all to be included if it means it has to be unplugged, I see `ts-node` supports using `@swc/wasm` now so that shouldn't be an issue though.